Replace node with blog in URL

Vandit1604 opened the pull request:
Replaced occurrences of `node` with `blog` in `content/` dir of blog pages.

krisstern commented:
Hi @Vandit1604, thanks for this PR. But would it be possible to add redirection from `node` to `blog` just in case?

Vandit1604 commented:
I can't remember if there's no URL with `node`. Perhaps I missed something. Could you please point to the URL that contains the word `node`?

krisstern commented:
There is no longer any reference of `node` in your implementation. But there could still be someone with the old URL, so we will need to be careful and redirect anyways while migrating from the old site.

Vandit1604 commented:
Checked Gatsby's documentation to see if they support redirections. I found this: https://www.gatsbyjs.com/docs/how-to/cloud/working-with-redirects-and-rewrites/#splat-redirects.

I implemented this in https://github.com/Vandit1604/jenkins-docs/pull/112/commits/17096b691c263d9e7cbcf34e5cef757cc5640110 but did not work. Furthermore, I believe it only works on Gatsby Cloud or Netlify.
I changed my approach. I'm looking into https://www.gatsbyjs.com/plugins/gatsby-plugin-htaccess/.

krisstern commented:
> Checked Gatsby's documentation to see if they support redirections. I found this: https://www.gatsbyjs.com/docs/how-to/cloud/working-with-redirects-and-rewrites/#splat-redirects.
> 
> I implemented this in [17096b6](https://github.com/Vandit1604/jenkins-docs/commit/17096b691c263d9e7cbcf34e5cef757cc5640110) but did not work. Furthermore, I believe it only works on Gatsby Cloud or Netlify. I changed my approach. I'm looking into https://www.gatsbyjs.com/plugins/gatsby-plugin-htaccess/.

@Vandit1604 Did you tried adding `force: true` to `createRedirect`?

Vandit1604 commented:
I'll try it

Vandit1604 commented:
It didn't work. Do you want me to commit the changes for you to take a look?

krisstern commented:
> It didn't work. Do you want me to commit the changes for you to take a look?

Sure

Vandit1604 commented:
```js
  createRedirect({
    fromPath: '/node/',
    toPath: '/blog/',
    exactPath: true,
    isPermanent: false,
    redirectInBrowser: true,
    force: true,
  });
  ```
  
  With this, Now the URL changes from `node` to `blog` but the page doesn't actually load or reload?
  I'll see if i can fix that and then commit my changes.

krisstern commented:
> ```js
> isPermanent: false,
> ```

How about changing `isPermanent: false,` to `isPermanent: true,` also?

Vandit1604 commented:
Nothing changed. The URL is changing to `/blog` but the page isn't reloading or loading.

I also tried https://github.com/gatsbyjs/gatsby/blob/master/examples/using-redirects/ but on redirects the page is not loading in that example too.

Vandit1604 commented:
I think this is the best we can do. if this doesn't work when we host the blog site we can use the .htacess file to define the redirect.

krisstern reviewed (approved):
LGTM

Vandit1604 commented on site/gatsby-node.js:
```diff
@@ -8,7 +8,25 @@ const { createFilePath } = require(`gatsby-source-filesystem`);
+    force: true,
+  });
+
+  createRedirect({
+    fromPath: `/node`,
+    toPath: `/node/`,
+    isPermanent: true,
+    redirectInBrowser: true,
+    force: true,
+  });
```
Here, We can redirect `/node` to `/blog/` here.

krisstern replied:
But does it take care of both `/node` and `/node/` redirecting to `/blog/`?
